### PR TITLE
Attempt to beautify About dialog on OS X.

### PR DESCRIPTION
**Pull request by endrift**
Added a 2x version of the logo, and cleaned up layout a bit. It still need a bit of work due to the way the dialog is constructed (the wxTextCtrl seem to be the wrong way to do the multiline text here), and this may break stuff on Windows; I didn't test it there.

Before:
screen shot 2015-04-25 at 4 11 23 am

After:
screen shot 2015-04-25 at 4 20 01 am

Note that the frame around that text box is from clicking on it. It goes away if you click somewhere else.

### CODE REVIEW

**Comment by Parlane** [comment]
@dolphin-emu-bot rebuild

**Comment by Parlane** [comment]
@endrift The reason they use a text field and not a label is so the user can select the text to copy/paste it :\

**Comment by BhaaLseN** [comment]
This looks terrible, why can't OSX just look the same as every other OS when we use a cross-platform GUI Toolkit :|

**Comment by RisingFog** [comment]
What a shame it looks so ugly on OSX. Also it appears that some of the changes breaks building on Windows.

**Comment by Pringo** [comment]
Yikes. Maybe this should have been addressed before #2016 got merged. Oh well.

Perhaps it would be best to remove the selectable text if this is going to be the result. It was never selectable in the previous version of the About screen and I don't remember users ever complaining about it.

**Comment by BhaaLseN** [comment]
I'm not overly happy about it, but maybe we should make them plain labels on OSX until we figure out a way of making them not suck?

**Comment by Pringo** [comment]
@BhaaLseN I agree with making those fields plain labels if this is going to be the result otherwise. I suspect that might also help with some of the formatting issues on other OS' as well.

**Comment by comex** [comment]
@BhaaLseN Because said toolkit sucks.

Here is a hack to remove the background on OS X while keeping the fields selectable.  Note that `wxNO_BORDER` must be on, and `wxTE_MULTILINE` must be off, even though there are multiple lines.  (Doing the same to MULTILINE controls is possible but requires slightly different code, since internally they're NSTextViews rather than NSTextFields, so I kept it simple.)

```
--- a/Source/Core/DolphinWX/AboutDolphin.cpp
+++ b/Source/Core/DolphinWX/AboutDolphin.cpp
@@ -17,10 +17,22 @@
 #include <wx/windowid.h>
 #include <wx/generic/statbmpg.h>
 
+#ifdef __APPLE__
+#import <AppKit/AppKit.h>
+#endif
+
 #include "Common/Common.h"
 #include "DolphinWX/AboutDolphin.h"
 #include "DolphinWX/resources/dolphin_logo.cpp"
 
+static void BanishBackground(wxTextCtrl* ctrl)
+{
+#ifdef __APPLE__
+       NSTextField* tf = (NSTextField*)ctrl->GetHandle();
+       tf.drawsBackground = NO;
+#endif
+}
+
 AboutDolphin::AboutDolphin(wxWindow *parent, wxWindowID id,
                const wxString &title, const wxPoint &position,
                const wxSize& size, long style)
@@ -47,9 +59,11 @@ AboutDolphin::AboutDolphin(wxWindow *parent, wxWindowID id,
        const wxString SupportText = _("Support");
 
        wxStaticText* const Dolphin = new wxStaticText(this, wxID_ANY, DolphinText);
-       wxTextCtrl* const Revision = new wxTextCtrl(this, wxID_ANY, RevisionText, wxDefaultPosition, wxDefaultSize);
+       wxTextCtrl* const Revision = new wxTextCtrl(this, wxID_ANY, RevisionText, wxDefaultPosition, wxDefaultSize, wxNO_BORDER | wxTE_NO_VSCROLL);
+       BanishBackground(Revision);
        wxStaticText* const Copyright = new wxStaticText(this, wxID_ANY, CopyrightText);
-       wxTextCtrl* const Branch = new wxTextCtrl(this, wxID_ANY, "", wxDefaultPosition, wxSize(300, 50), wxTE_MULTILINE | wxNO_BORDER | wxTE_NO_VSCROLL);
+       wxTextCtrl* const Branch = new wxTextCtrl(this, wxID_ANY, "", wxDefaultPosition, wxSize(300, 50), wxNO_BORDER | wxTE_NO_VSCROLL);
+       BanishBackground(Branch);
        wxStaticText* const Message = new wxStaticText(this, wxID_ANY, Text);
        wxStaticText* const Update = new wxStaticText(this, wxID_ANY, CheckUpdateText);
        wxStaticText* const FirstSpacer = new wxStaticText(this, wxID_ANY, wxString("  |  "));
```

**Comment by endrift** [comment]
Ugh ok, here are some fixes. After digging through wxWidget's headers and source *manually* it seems that this is only supported on wxMac. It turns out that the API is so ludicrously bad that that function only even exists in wxMac. Who designed this garbage library?

**Comment by lioncash** [comment]
Actually in an updated head version of the repo, there has been a function implemented to get the scale factor.

**Comment by endrift** [comment]
Well, there is CreateScaled or whatever it was, but it's completely dummied out in every other version of wxWidgets. Presumably that means setting the contents after-the-fact would have them be too big, but it also looks like you can't set a wxImage anywhere other than the constructor, so it's irrelevant. What is this offal? Who thought any of this was a good idea?

**Comment by lioncash** [comment]
Hmm, yeah, just checked, I guess the commit I read a while ago got reverted or something. The API has always been garbage in terms of flexibility, so I'm not surprised it would be a pain.

File: Source/Core/DolphinWX/AboutDolphin.cpp [hunk 1]
```diff
@@ -17,19 +17,47 @@
...
 #include "Common/Common.h"
 #include "DolphinWX/AboutDolphin.h"
 #include "DolphinWX/resources/dolphin_logo.cpp"
 
+static void BanishBackground(wxTextCtrl* ctrl)
+{
+#ifdef __APPLE__
+       NSTextField* tf = (NSTextField*)ctrl->GetHandle();
+       tf.drawsBackground = NO;
+#endif
```
Comment (lioncash) [inline Source/Core/DolphinWX/AboutDolphin.cpp]: Have you looked into just setting a wxTextAttr before doing this? It allows you to set background color.

**Comment by Rukario** [comment]
I think "should not be used to play games you do not legally owned" shouldn't be there. It will give users the 'idea'.

File: Source/Core/DolphinWX/AboutDolphin.cpp (hunk 1 above)
Comment (endrift) [inline Source/Core/DolphinWX/AboutDolphin.cpp]: The idea is not to set the background color, but to remove it entirely. Setting it to wxTransparentColour didn't remove it; it's still white.

**Comment by Parlane** [comment]
@dolphin-emu-bot rebuild

**Comment by bb010g** [comment]
[**@Rukario**](https://github.com/Rukario) commented on [Apr 25, 2015, 3:32 PM PDT](https://github.com/dolphin-emu/dolphin/pull/2324#issuecomment-96288827 "2015-04-25T22:32:29Z - Replied by Github Comment Enhancer"):
> I think "should not be used to play games you do not legally owned" shouldn't be there. It will give users the 'idea'.

People'll learn about piracy from other places. We want to be covered.

**Comment by mbc07** [comment]
Well, multiline text is broken on Windows, not sure if it has something related to my laptop using a custom DPI but since this branch got merged I get this:
capturar2

Before I would get proper multiline text:
capturar

**Comment by Parlane** [comment]
Ye comex removed wxTE_MULTILINE for all platforms :\

**Comment by mbc07** [comment]
Well, I know it's a minor detail but it would be nice having multiline text working again on Windows. I'm not totally sure but a simple if/else would suffice, wouldn't?

**Comment by RisingFog** [comment]
ok, I'll take a better look at this tomorrow to try and reach a compromise for all platforms.

I'm thinking perhaps having it when you click on the textbox, it copies everything within said box. Thoughts?

**Comment by mbc07** [comment]
Well, at least for me these kind of text fields can get annoying sometimes: if I want to select only the revision hash to paste in a GitHub URL, for example, in a text field like that it would automatically copy all text and I would need an additional step like pasting the text in notepad to then select only hash. However, this may be extremely helpful for a newbie asking for help in the forums, so I'm neutral here, +0/-0 from me...

**Comment by comex** [comment]
just put multiline back and I'll fix BanishBackground.

**Comment by Rukario** [comment]
@bb010g I suppose we can rewrite it rather than removing it altogether in defense of piracy. We can expand "Dolphin is blahblah" with additional sentences that it does not come with any games and/or homebrew apps. Then explain how to obtain them in correct and legal way (something along with real Wii or special cd drive capable of reading Wii discs is enough), and state that it's not responsible for any other methods, such as pirating (<- not mention it). This way, users will not get the idea.

**Comment by bb010g** [comment]
@Rukario IIRC, we already discuss all that on the Dolphin wiki and FAQ. They go to the Dolphin website to get the emulator anyway; if the legal methods need to be more apparent, that is where they should be, not in an about dialog. I don't see a problem as it is right now, but I may be wrong. Anyone else's thoughts?

**Comment by endrift** [comment]
Not really sure this is a good place to be discussing this in the first place, but in my opinion, the whole line is cheesy and unnecessary. People who have downloaded it and are looking at the about screen are fully aware of what can and what should (and what should not) be done with it. It's not going to give someone ideas one way or the other that far into the process.

**Comment by RisingFog** [comment]
@comex does banishbackground also work for linux?

**Pull request merged**